### PR TITLE
[IDB IGNORE] [MDB IGNORE] Makes the icons/mob folder sane

### PR DESCRIPTION
**Pull request by andrea-feather**
## About The Pull Request

Reorganizes the entire `icons/mob` folder.

Added the following new subfolders:
- `nonhuman-player` (this was initially just called "antag", but then I realized guardians aren't technically antags)
- `simplemob`
- `silicon`
- `effects` (for bloodstains, fire, etc)
- `simplemob/held-pets` (for exactly that -- I wasn't sure if this should go in inhands instead)
- `species/monkey`

Moves the following stuff:
- All human parts moved into `species`, with moth, lizard, monkey, etc parts moved to corresponding subfolders. Previously, there were _some_ moth parts in `mob/species/moth`, and others just loose in `mob`. Other species were similar.
- `icemoon`, `lavaland`, and `jungle` folders made into subfolders of `simplemob`
- All AI and silicon stuff, as well as Beepsky et al. into the `silicon` folder, simplemobs into the `simplemob` folder, aliens into the `nonhuman-player` folder, etc.
- Split up `animal_parts.dmi` into two `bodyparts.dmi` which were put in their respective folders (`species/alien` and `species/monkey`)

Code changes:
- Filepath changes to account for all of this
- Adds a check when performing surgery on monkeys and xenos, because we can no longer assume their limbs are in the same file
- Turns some hardcoded statues and showcases that were built into maps into objects instead

Things I'd like to do in the future but cant be assed right now:
- Remove primarily-antag sprites from `simplemob/mob.dmi` (Revenant, Morph, etc.) and put them in the `nonhuman-player` folder
- Split up `mutant_bodyparts.dmi` into different files for Tizirans, Felinids, monkeys, etc and put them in their own folders. Those may have once been meant primarily for mutated humans but that's now how they're being used right now.

## Why It's Good For The Game

That folder was a fucking disaster and it hurt my soul to look at it.

I'm sure there will be some disagreements about how exactly I organized it all (should bots really go in `silicon`, or do they belong in `simplemobs`?), but the original folder had so many things dumped into a single big folder and it was a pain in the ass to find anything, _especially_ when it came to humanoids. Having some level of organization will make future changes to organization easier as well.

## Changelog

:cl:
refactor: Reorganized the icons/mob folder and added a bunch of new subfolders to make finding things easier.
/:cl:

### CODE REVIEW

**Comment by skylord-a52** [comment]
ah shit i fucked up a merge conflict slightly and now it's not looking for the right files, one second....

**Review by Kapu1178 — CHANGES_REQUESTED** [review]
Not a fan of `/species` >> `/humanoid`, feels less descriptive and not as accurate.

**Comment by skylord-a52** [comment]
> Not a fan of `/species` >> `/humanoid`, feels less descriptive and not as accurate.

I guess. To me "species" was a bit confusing (species of _what?_), but I'll see what others think.

**Comment by Mothblocks** [comment]
nit: simplemob should be simple

**Comment by Kapu1178** [comment]
> > Not a fan of `/species` >> `/humanoid`, feels less descriptive and not as accurate.
> 
> I guess. To me "species" was a bit confusing (species of _what?_), but I'll see what others think.

species is a datum that anyone familiar with dna/human code will be aware of.

**Comment by skylord-a52** [comment]
> also, keep _**ALL**_ bodyparts in the same root folder.

Sorry, can you clarify? If you mean human species bodyparts, those were already split up into separate subfolders.

**Comment by Kapu1178** [comment]
> > also, keep _**ALL**_ bodyparts in the same root folder.
> 
> Sorry, can you clarify? If you mean human species bodyparts, those were already split up into separate subfolders.

Anything that is an icon for an `/obj/item/bodypart` should be in the same root folder.

**Comment by skylord-a52** [comment]
> Anything that is an icon for an `/obj/item/bodypart` should be in the same root folder.

I... don't fully understand or agree. I think it makes much more sense to separate by species so that a species' bodyparts and external organs are in the same place. If I'm understanding you correctly, what you want is for all bodyparts to be piled within the same folder, with only external organs separated by species. That seems somewhat messy and inconsistent.

But I can add a commit in a few days once I'm back to my computer.

What's your reasoning behind this? If it's a concern of being able to programmatically select them, we could instead continue the existing format of `species/[species]/bodyparts.dmi`, and bodyparts could be accessed programmatically that way. Or, well, I guess not for aliens, but... maybe that's what you meant? That alien bodyparts should follow the same pattern?

**Comment by Kapu1178** [comment]
> > Anything that is an icon for an `/obj/item/bodypart` should be in the same root folder.
> 
> I... don't fully understand or agree. I think it makes much more sense to separate by species so that a species' bodyparts and external organs are in the same place. If I'm understanding you correctly, what you want is for all bodyparts to be piled within the same folder, with only external organs separated by species. That seems somewhat messy and inconsistent.
> 
> But I can add a commit in a few days once I'm back to my computer.
> 
> What's your reasoning behind this? If it's a concern of being able to programmatically select them, we could instead continue the existing format of `species/[species]/bodyparts.dmi`, and bodyparts could be accessed programmatically that way. Or, well, I guess not for aliens, but... maybe that's what you meant? That alien bodyparts should follow the same pattern?

I said ROOT folder. 
`/species/[species]/bodyparts.dmi` is the desired place for bodypart DMIs.
Xenos are in a weird state because they are carbon, but dont have a species. they are a `shitcode outlier` and should be put in `/species` anyway because of this.

**Comment by skylord-a52** [comment]
> I said ROOT folder. `/species/[species]/bodyparts.dmi` is the desired place for bodypart DMIs. Xenos are in a weird state because they are carbon, but dont have a species. they are a `shitcode outlier` and should be put in `/species` anyway because of this.

Right! Thanks, that's what I was confused about. Will do.

File: _maps/map_files/CTF/limbo.dmm
Comment (san7890) [inline _maps/map_files/CTF/limbo.dmm:900]: We have the fucking icon var_edited? Fuck!

Do you think you'd be willing to create a subtype of /obj/structure/showcase with these exact variables applied? This same thing shows up in a map key somewhere else in the map, as well as in CentCom.dmm. Then, replace these instances completely with the new subtype? Let me know if you need help figuring out how to do this, it's a great step to ensuring that if we want to change around icons again, we don't have to touch maps (we also shouldn't have done this in the first place).

There's also the statue that you change in IceBoxStation.dmm. That's also a good candidate for the subtype.

Comment (skylord-a52) [reply _maps/map_files/CTF/limbo.dmm]: Done -- I'll need a review of that stuff, since I don't really know what I'm doing when it comes to map edits.

**Comment by skylord-a52** [comment]
Alright, all requests have been handled, going to see if I can resolve that merge conflict now.

**Comment by skylord-a52** [comment]
why on earth do i have a merge conflict when pushing to tg but i DIDNT get one when squashing and merging tg to my branch... git is pain.

**Comment by LemonInTheDark** [comment]
You uh, may wanna walk that one back a few steps

**Comment by skylord-a52** [comment]
So. After some testing, I've found that if I make a personal branch that's a copy of tgstation/tgstation and then try to merge my changes from this branch onto it, there's no merge conflict. The conflict only appears in the PR on the website, for some reason. I have no idea what's going on here.

**Comment by LemonInTheDark** [comment]
Your local copy of tgstation is out of date with master, which is why it only shows here.
That or you messed up a merge commit, and git has decided to bring us all down swinging.

If you'd like help unfucking your repo, feel free to ask in coding general, or dm me directly on discord at TheFinalPotato#9874 if you can't for some reason or another.

Essentially, you're gonna want to walk back to before you did the merge, and then redo it. I think?

File: _maps/map_files/Mafia/mafia_spiderclan.dmm
Comment (san7890) [inline _maps/map_files/Mafia/mafia_spiderclan.dmm:94]: Shouldn't this be `/obj/structure/showcase/katana` instead of the base showcase? Looks good otherwise.

Comment (skylord-a52) [reply _maps/map_files/Mafia/mafia_spiderclan.dmm:94]: Thanks for catching that

**Comment by skylord-a52** [comment]
I did it! I unfucked the branch! Hooray!

**Comment by san7890** [comment]
I don't think IDB Checks will work on this so you might wanna put [IDB IGNORE] in the title of your PR. We're not changing the actual sprites, just shuffling a buncha files around, so it's probably for the best.

**Comment by skylord-a52** [comment]
What do I need to do for this to get merged? I don't think those last few tests are ever going to be positive, just because so much is being changed.

**Comment by san7890** [comment]
> What do I need to do for this to get merged? I don't think those last few tests are ever going to be positive, just because so much is being changed.

You fundamentally can't hit re-run checks as a maintainer on IDB2 because of some bullshit that they explained to me once in coderbus, so sending a new commit will result in your IDB IGNORE being honored. It's not so much the checks as it is someone finally mustering up the time/courage to go through and ensure they aren't going to break stuff if this PR is merged. If it goes stale or if you want to generate discussion about this, I think either #coding-general or #spriting-general in the tgstation Discord wouldn't be a bad idea to generate more interest.

**Comment by skylord-a52** [comment]
> It's not so much the checks as it is someone finally mustering up the time/courage to go through and ensure they aren't going to break stuff if this PR is merged. If it goes stale or if you want to generate discussion about this, I think either #coding-general or #spriting-general in the tgstation Discord wouldn't be a bad idea to generate more interest.

Ah okay, that's fine then! Just wanted to make sure everything on my end was good.

**Comment by dragomagol** [comment]
I'll take a look at it, you've just got one merge conflict to clear up in the meantime

File: code/modules/mob/living/basic/vermin/mothroach.dm
Comment (dragomagol) [inline code/modules/mob/living/basic/vermin/mothroach.dm:8]: Should go in inhands

File: code/modules/mob/living/basic/vermin/mothroach.dm
Comment (dragomagol) [inline code/modules/mob/living/basic/vermin/mothroach.dm:10]: Would also argue this should go in mob/clothing/head

File: code/modules/mob/living/living_defines.dm
Comment (dragomagol) [inline code/modules/mob/living/living_defines.dm:188]: Same as with mothroaches

File: code/modules/mob/living/living_defines.dm
Comment (dragomagol) [inline code/modules/mob/living/living_defines.dm:192]: Ditto

File: code/modules/pai/pai.dm
Comment (dragomagol) [inline code/modules/pai/pai.dm:7]: More inhands

File: code/modules/pai/pai.dm
Comment (dragomagol) [inline code/modules/pai/pai.dm:9]: And head clothing

File: code/modules/surgery/bodyparts/_bodyparts.dm
Comment (dragomagol) [inline code/modules/surgery/bodyparts/_bodyparts.dm:735]: Does this comment need to be here?

File: code/modules/surgery/bodyparts/_bodyparts.dm
Comment (dragomagol) [inline code/modules/surgery/bodyparts/_bodyparts.dm:729]: Does this comment need to be here?

**Review by dragomagol — CHANGES_REQUESTED** [review]
Mostly just a little shuffling around for inhands

File: code/modules/surgery/bodyparts/_bodyparts.dm
Comment (Mothblocks) [reply code/modules/surgery/bodyparts/_bodyparts.dm:735]: no

File: code/modules/surgery/bodyparts/_bodyparts.dm
Comment (Mothblocks) [reply code/modules/surgery/bodyparts/_bodyparts.dm:729]: no historical comments

Comment (skylord-a52) [reply code/modules/surgery/bodyparts/_bodyparts.dm]: removed

File: code/modules/surgery/bodyparts/_bodyparts.dm
Comment (skylord-a52) [reply code/modules/surgery/bodyparts/_bodyparts.dm]: removed

File: code/modules/mob/living/basic/vermin/mothroach.dm
Comment (skylord-a52) [reply code/modules/mob/living/basic/vermin/mothroach.dm]: done (as well as all the other comments)

**Comment by skylord-a52** [comment]
the error in all those tests does not appear on my local branch. it should not be POSSIBLE on either local or remote branch -- line 505 has nothing to do with `icons/mob/onfire.dmi`, and that filepath should not appear anywhere in the code (it doesnt on local). force push and force pull do nothing, claiming my local and remote are identical. i'm in so much pain.

**Pull request merged**
